src/components/ButtonWithLoading/index.js

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Button } from "../Button";
+import Loader from "./Loader";
+
+const ButtonWithLoading = ({ isLoading, children, ...props }) => {
+  const { variant } = props;
+
+  return (
+    <Button {...props}>
+      <Loader isLoading={isLoading} variant={variant}>
```

Comment:
A possible workaround would be to return the children if loading is set to false without adding the custom style.

Reply:
Updated using conditional rendering and `visibility: hidden` to save the measurements of Button